kratos/utilities/point_locator.cpp

```diff
@@ -0,0 +1,148 @@
+    {
+        int found_node_id = -1; // if no node is found this will be returned
+        bool is_close_enough = false;
+
+        int global_nodes_found = 0;
+
+        // note that this cannot be omp bcs breaking is not allowed in omp
+        for (auto& r_node : mrModelPart.GetCommunicator().LocalMesh().Nodes())
+        {
+            is_close_enough = NodeIsCloseEnough(r_node, rThePoint, DistanceThreshold);
+            if (is_close_enough)
+            {
+                global_nodes_found = 1;
+                found_node_id = r_node.Id();
+                break;
+            }
+        }
+
+        CheckResults("Node", rThePoint, global_nodes_found);
+
+        return found_node_id;
+    }
+
+    int PointLocator::FindElement(const Point& rThePoint,
+                                  Vector& rShapeFunctionValues) const
+    {
+        int found_element_id = -1; // if no element is found this will be returned
+        const auto& r_elements = mrModelPart.GetCommunicator().LocalMesh().Elements();
+        FindObject(r_elements, "Element",
+                   rThePoint, found_element_id,
+                   rShapeFunctionValues);
+        return found_element_id;
+    }
+
+    int PointLocator::FindCondition(const Point& rThePoint,
+                                    Vector& rShapeFunctionValues) const
+    {
+        int found_condition_id = -1; // if no condition is found this will be returned
+        const auto& r_conditions = mrModelPart.GetCommunicator().LocalMesh().Conditions();
+        FindObject(r_conditions, "Condition",
+                   rThePoint, found_condition_id,
+                   rShapeFunctionValues);
+        return found_condition_id;
+    }
+
+    template<typename TObjectType>
+    void PointLocator::FindObject(const TObjectType& rObjects,
+                                  const std::string& rObjectName,
+                                  const Point& rThePoint,
+                                  int& rObjectId,
+                                  Vector& rShapeFunctionValues) const
+    {
+
+        const int domain_size = mrModelPart.GetProcessInfo()[DOMAIN_SIZE];
+
+        const auto& r_geom = rObjects.begin()->GetGeometry();
+
+        KRATOS_ERROR_IF_NOT(static_cast<std::size_t>(domain_size) == r_geom.WorkingSpaceDimension())
+            << "Domain size (" << domain_size << ") and WorkingSpaceDimension ("
+            << r_geom.WorkingSpaceDimension() << ") of the " << rObjectName
+            << " are not equal!" << std::endl;
+
+        bool is_inside;
+
+        int global_objects_found = 0;
+        array_1d<double, 3> local_coordinates;
+
+        // note that this cannot be omp bcs breaking is not allowed in omp
+        for (auto& r_object : rObjects)
+        {
+            is_inside = r_object.GetGeometry().IsInside(rThePoint, local_coordinates);
+            if (is_inside)
+            {
+                global_objects_found = 1;
```

Comment:
Should this be += 1 ?
otherwise checks in the CheckResults (global_objects_found >1) will be not working.

Reply:
no here it is correct the way it is

the name is misleading, it is actually the local number of found entities (cannot be larger than 1 bcs I break after one is found)

In `CheckResults` I do compute the global number then (line 116)

Reply:
I see. how about changing the name to local_objects_found ? and defining a different variable global_objects_found in line 116 ? Just to be more clear in implementation.

Reply:
will do :+1: